pkg/syncer/syncer.go

```diff
@@ -136,7 +143,16 @@ func StartSyncer(ctx context.Context, cfg *SyncerConfig, numSyncerThreads int, i
 	upstreamConfig.Host = syncerVirtualWorkspaceURL
 	rest.AddUserAgent(upstreamConfig, "kcp#spec-syncer/"+kcpVersion)
 
-	upstreamDynamicClusterClient, err := kcpdynamic.NewForConfig(upstreamConfig)
+	upstreamSyncerClusterClient, err := kcpdynamic.NewForConfig(upstreamConfig)
+	if err != nil {
+		return err
+	}
+
+	upstreamUpsyncConfig := rest.CopyConfig(cfg.UpstreamConfig)
+	// Upsyncing to Virtual Workspace for upsyncing
+	upstreamUpsyncConfig.Host = strings.Replace(syncerVirtualWorkspaceURL, "/services/syncer", "/services/upsyncer", 1)
+	upstreamUpsyncConfig.UserAgent = "kcp#upsyncer/" + kcpVersion
```

Comment:
would prefer this to be the binary "syncer", and then something to qualify it as upsyncer controller.

Reply:
Fixed in commit da89d306186c852a5f9f9ab2a3a549f896f5f7e0